### PR TITLE
fix: detect number first before check NaN

### PR DESCRIPTION
**Pull request by fengmk2**

### CODE REVIEW

File: lib/object.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ exports.getOwnEnumerables = function getOwnEnumerables(obj, ignoreNull) {
   return Object.keys(obj).filter(function(key) {
     if (ignoreNull) {
       var value = obj[key];
-      if (value === null || value === undefined || isNaN(value)) {
+      if (value === null || value === undefined || (typeof value === 'number' && isNaN(value))) {
```
Comment (dead-horse) [inline lib/object.js]: 可以直接用 `Number.isNaN()`

Comment (dead-horse) [reply lib/object.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

The Number.isNaN() method determines whether the passed value is NaN and its type is Number. It is a more robust version of the original, global isNaN().

Comment (fengmk2) [reply lib/object.js]: 难道不是一样的？

Comment (fengmk2) [reply lib/object.js]: 果然不一样。。。

**Comment by fengmk2** [comment]
1.16.1

**Pull request merged**